src/OpenTelemetry.Api/.publicApi/Stable/PublicAPI.Unshipped.txt

```diff
@@ -0,0 +1,7 @@
+abstract OpenTelemetry.Logs.LoggerProviderBuilder.AddInstrumentation<TInstrumentation>(System.Func<TInstrumentation>! instrumentationFactory) -> OpenTelemetry.Logs.LoggerProviderBuilder!
```

Comment:
Just curious, When would `AddInstrumentation` be needed with respect to logging?

Reply:
Whenever the instrumentation libraries start producing logs. As of today, only traces and metrics are emitted by most, but there is nothing preventing an instrumentation library to emit logs too. Of course, it requires an `ILoggerFactory` or an `ILogger`  to be given to the instrumentation library (unlike Meter,ActivitySource, which can be created out of thin air using ctor() )

Reply:
Got it - I was curious to know if there are any specific examples where this is needed today.

Reply:
no specific example I am aware of. Most libraries are natively producing Ilogger logs today, so there wasn't a pressing demand for an instrumentation library for logs.

Reply:
The other nice thing `AddInstrumentation` does is allow you to attach things to the lifetime of a provider. When disposed the provider will in turn call `Dispose` on anything added as instrumentation (if it implements `IDisposable`).

I used that way back in the [EventSource -> Logs PoC](https://github.com/open-telemetry/opentelemetry-dotnet/blob/5be8226861c12b021bc1ea65516ea05755e8c1f9/src/OpenTelemetry.Extensions.EventSource/OpenTelemetryEventSourceLoggerProviderBuilderExtensions.cs#L78-L83).

Probably won't be used a whole lot but good to keep logs/metrics/tracing providers consistent in that way?

Reply:
Sounds good to me 👍